src/test/java/org/apache/commons/imaging/examples/tiff/ReadAndRenderFloatingPoint.java

```diff
@@ -0,0 +1,173 @@
+        // Render the first directory in the file
+        if (!directory.hasTiffImageData()) {
+            System.err.println("First directory in file does not have image");
+            System.exit(-1);
+        }
+
+        // Obtain metadata about what kind of data is in the product.
+        // Because this demo is designed for floating-point data,
+        // it will expect the sample format to include only one element
+        // of type floating point.   Beyond that, the TIFF specification
+        // allows a large number of variations in format (16, 24, or 32 byte
+        // floats; tiles versus strips; etc.).  Unfortunately the test data
+        // for the less common variations was not available when Commons Imaging
+        // was implemented and so the API will not be able to handle all
+        // of them.  This test will simply allow the API to throw an exception
+        // if an unsupported format is encountered.
+        //    The getFieldValue call allows an application to provide a
+        // boolean indicating that the field must be present for processing
+        // to continue. If it does not, an exception is thrown.
+        short[] sampleFormat = directory.getFieldValue(TiffTagConstants.TIFF_TAG_SAMPLE_FORMAT, true);
+        short samplesPerPixel = directory.getFieldValue(TiffTagConstants.TIFF_TAG_SAMPLES_PER_PIXEL);
+        short[] bitsPerPixel = directory.getFieldValue(TiffTagConstants.TIFF_TAG_BITS_PER_SAMPLE, true);
+        if (sampleFormat[0] != TiffTagConstants.SAMPLE_FORMAT_VALUE_IEEE_FLOATING_POINT) {
+            System.err.println("This example requires a data source with a floating-point format");
+            System.exit(-1);
+        }
+
+        System.out.print("Bits per pixel: ");
+        for (int i = 0; i < samplesPerPixel; i++) {
+            System.out.format("%s%d", i > 0 ? ", " : "", bitsPerPixel[i]);
+        }
+        System.out.println("");
+
+        // Create a TIFF Photometric Interpreter to perform a grayscale
+        // rendering.  A Photometric Interpreter is a class that maps a
+        // floating-point value to a pixel (RGB) equivalent.
+        // Because there is no way to know the range of
+        // values in the TIFF before it is read.  So the special-purpose
+        // interpreter permits an application to extract the
+        // min and maximum bounds of the data as an auxiliary function
+        // of interpreting the data. Unfortunately, if we wish for a
+        // good interpretation, we need to read the data twice.
+        //    For this demo, we store the Photometric Interpreter instance
+        // as a option-parameter to be passed into the read-image method.
+        PhotometricInterpreterFloat pi = new PhotometricInterpreterFloat(0.0f, 1.0f);
+        HashMap<String, Object> params = new HashMap<>();
+        params.put(TiffConstants.PARAM_KEY_CUSTOM_PHOTOMETRIC_INTERPRETER, pi);
+        System.out.println("Reading image");
```

Comment:
Unless really necessary, it would be better to remove the string printed to stdout, as they can get in the way when troubleshooting failing tests.

Reply:
No problem.   I will remove it with the next commit.   

I used in in the example application because I was trying to judge how much time different operations require.  But it is no longer needed.